dianna/utils/maskers.py

```diff
@@ -1,45 +1,58 @@
+import heapq
 import warnings
 from typing import Union
 import numpy as np
+from numpy import ndarray
 from skimage.transform import resize
 
 
-def generate_masks(input_data: np.array,
-                   number_of_masks: int,
-                   p_keep: float = 0.5):
+def generate_masks(
+    input_data: np.array,
+    number_of_masks: int,
+    feature_res: int = 8,
+    p_keep: float = 0.5,
+):
     """Generate masks for time series data given a probability of keeping any time step or channel unmasked.
 
     Args:
         input_data: Timeseries data to be explained.
```

Comment:
```suggestion
        input_data: Timeseries data to be masked.
```

Reply:
I see that the shape information is obtained by hardcoding the index in many places in the code for example  `number_of_channels = input_data.shape[1]`. Shall we add a text in the docstring what is the expected shape of `input_data`?